homeassistant/components/sleepiq/number.py

```diff
@@ -215,15 +215,17 @@ async def async_setup_entry(
                 )
             )
         for core_climate in bed.foundation.core_climates:
-            entities.append(
-                SleepIQNumberEntity(
```

Comment:
Why don't we add it as an optional field to the constructor here - that would simplify the code quite a bit

Reply:
Runs into the same FrozenInstanceError. 

Init: 
```
 for core_climate in bed.foundation.core_climates:
            entities.append(
                SleepIQNumberEntity(
                    data.data_coordinator,
                    bed,
                    core_climate,
                    NUMBER_DESCRIPTIONS[CORE_CLIMATE_TIMER],
                    core_climate.max_core_climate_time,
                )
            )
```

Class:

```
class SleepIQNumberEntity(SleepIQBedEntity[SleepIQDataUpdateCoordinator], NumberEntity):
    """Representation of a SleepIQ number entity."""

    entity_description: SleepIQNumberEntityDescription
    _attr_icon = "mdi:bed"

    def __init__(
        self,
        coordinator: SleepIQDataUpdateCoordinator,
        bed: SleepIQBed,
        device: Any,
        description: SleepIQNumberEntityDescription,
        native_max_value: int | None = 360,
    ) -> None:
        """Initialize the number."""
        description.native_max_value = native_max_value
        self.entity_description = description
        self.device = device

        self._attr_name = description.get_name_fn(bed, device)
        self._attr_unique_id = description.get_unique_id_fn(bed, device)
        if description.icon:
            self._attr_icon = description.icon

        super().__init__(coordinator, bed)

    @callback
    def _async_update_attrs(self) -> None:
        """Update number attributes."""
        self._attr_native_value = float(self.entity_description.value_fn(self.device))

```


Error:

```
2024-01-28 23:48:35.733 ERROR (MainThread) [homeassistant.components.number] Error while setting up sleepiq platform for number
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/helpers/entity_platform.py", line 326, in _async_setup_platform
    await asyncio.shield(task)
  File "/workspaces/core/homeassistant/components/sleepiq/number.py", line 192, in async_setup_entry
    SleepIQNumberEntity(
  File "/workspaces/core/homeassistant/components/sleepiq/number.py", line 246, in __init__
    description.native_max_value = native_max_value
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 4, in __setattr__
dataclasses.FrozenInstanceError: cannot assign to field 'native_max_value'
```

Reply:
Just put  `self._attr_native_value = native_max_value` in the constructor.  Should also default it to None and check for that so it doesn't override all the other max values

Reply:
Thank you so much for your feedback and help here. I got it all working! Let me know if there is anything else I can fix up. 

Again, I really appreciate your help.